src/gui/mzroll/ligandwidget.cpp

```diff
@@ -514,6 +514,67 @@ void LigandWidget::saveCompoundList(){
     }
 }
 
+void LigandWidget::saveCompoundListToPolly(QString fileName,QString dbname){
```

Comment:
@Ky5595 @shubhra-agrawal Why are we doing this? Why can't just take the file from user's system?

Reply:
@sahil21  Basically, we want to upload the compound database that was used to generate the peak table, to Polly. We could have had a mapping of the compound databases present in the ligand widget and their actual location on the system.
But, To validate it before uploading, we had to write a function separately. Personally, I feel that will add one more redundant function to the code, so i used already existing functions from Elmaven code.

This takes care of the following scenarios -
1. If the user uploads some compound dataset to Elmaven, does some analysis and for some reason that file is deleted from the system.
If we use fixed mapping, Elmaven will crash. 
2. If the file is modified on the system, Elmaven will upload the file but further workflow on Polly will break.

The current code takes care of both these scenarios by writing to a temporary file from RAM (Elmaven session variables) and then uploading it to Polly.

Reply:
@Ky5595 saveCompoundList & saveCompoundListToPolly is sharing around 95% of code. Please refactor this.